src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +166,104 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             }
         }));
     }
+
+    private async Task SubscribeDashboardLogsAsync()
+    {
+        string? dashboardResourceId = null;
+
+        try
+        {
+            using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1));
+            await foreach (var @event in resourceNotificationService.WatchAsync(cts.Token))
+            {
+                // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                // When the resource id is not the same as the resource name, it's the replica.
+                if (@event.Resource.Name == KnownResourceNames.AspireDashboard && @event.ResourceId != @event.Resource.Name)
+                {
+                    dashboardResourceId = @event.ResourceId;
+                    break;
+                }
+            }
+        }
+        catch (OperationCanceledException)
+        {
+            distributedApplicationLogger.LogError("Failed to get the dashboard logs.");
+            return;
+        }
+
+        if (dashboardResourceId is null)
+        {
+            return;
+        }
+
+        var defaultDashboardLogger = loggerFactory.CreateLogger("Aspire.Hosting.Dashboard");
+
+        var loggerCache = new Dictionary<string, ILogger>();
+
+        try
+        {
+            // We turned on the JSON formatter for the logger, so we can log the log lines as JSON.
+            await foreach (var batch in resourceLoggerService.WatchAsync(dashboardResourceId).ConfigureAwait(false))
+            {
+                foreach (var logLine in batch)
+                {
+                    try
+                    {
+                        var logMessage = JsonSerializer.Deserialize(logLine.Content, DashboardLogMessageContext.Default.DashboardLogMessage);
+
+                        if (logMessage is not null)
+                        {
+                            var logLevel = logMessage.LogLevel switch
+                            {
+                                "Information" => LogLevel.Information,
+                                "Debug" => LogLevel.Debug,
+                                "Warning" => LogLevel.Warning,
+                                "Error" => LogLevel.Error,
+                                "Critical" => LogLevel.Critical,
+                                "Trace" => LogLevel.Trace,
+                                _ => LogLevel.Information
+                            };
+
+                            if (!loggerCache.TryGetValue(logMessage.Category, out var logger))
+                            {
+                                var categoryTrimmed = logMessage.Category.StartsWith("Aspire.Dashboard.") ?
+                                    logMessage.Category["Aspire.Dashboard.".Length..] : logMessage.Category;
```

Comment:
Looks strange to see Aspire.Hosting.Dashboard.Aspire.Dashboard.Category, so I trim the prefix and append Aspire.Hosting. Why is this important? Well there are logs emitting from categories that don't start with Aspire.Dashboard so we want to prefix *all* logs so that they can be controlled by config.

Reply:
> Well there are logs emitting from categories that don't start with Aspire.Dashboard so we want to prefix all logs so that they can be controlled by config.

This would make a great comment in the code.

Reply:
Done